lib/IRGen/GenEnum.cpp

```diff
@@ -6886,13 +6919,19 @@ SingletonEnumImplStrategy::completeEnumTypeLayout(TypeConverter &TC,
       auto alignment = fixedEltTI.getFixedAlignment();
       applyLayoutAttributes(TC.IGM, theEnum, /*fixed*/true, alignment);
 
+      IsABIAccessible_t isABIAccessible = IsABIAccessible;
+      if (Type.getASTType()->isNoncopyable() &&
+          !IGM.getSILModule().isTypeMetadataAccessible(Type.getASTType()))
```

Comment:
This seems like not quite the right predicate.  Surely we can have a non-copyable type with inaccessible type metadata that's still ABI-accessible — maybe it doesn't have a `deinit` itself and we can still meaningfully recurse through this level of the type.  Even if this is currently impossible for some reason, it feels like something we should have a common function to test rather than repeating the assumption in many places.